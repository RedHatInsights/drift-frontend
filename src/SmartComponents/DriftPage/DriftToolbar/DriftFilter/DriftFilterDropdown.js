import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core';
import { FilterIcon } from '@patternfly/react-icons';

function DriftFilterDropdown(props) {
    const { filterType, toggleFilterType } = props;
    const [ isOpen, toggleOpen ] = useState(false);

    const selectFilter = (type) => {
        toggleFilterType(type);
        toggleOpen(!isOpen);
    };

    const dropdownItems = [
        <DropdownItem
            data-ouia-component-id='fact-name-filter'
            key='fact-name'
            onClick={ () => selectFilter('Fact name') }
        >
            Fact name
        </DropdownItem>,
        // Using this code block for the next PR.
        /*<DropdownItem
            data-ouia-component-id='fact-type-filter'
            key='fact-type'
            onClick={ () => selectFilter('Fact type') }
        >
            Fact type
        </DropdownItem>,*/
        <DropdownItem
            data-ouia-component-id='state-filter'
            key='state'
            onClick={ () => selectFilter('State') }
        >
            State
        </DropdownItem>
    ];

    return (
        <Dropdown
            ouiaId='drift-filter-dropdown'
            className='comparison-filter-dropdown-width'
            toggle={ <DropdownToggle
                onToggle={ toggleOpen }
                ouiaId='drift-filter-toggle'
                icon={ <FilterIcon /> }>
                { filterType }
            </DropdownToggle> }
            isOpen={ isOpen }
            dropdownItems={ dropdownItems }
        />
    );
}

DriftFilterDropdown.propTypes = {
    filterType: PropTypes.string,
    toggleFilterType: PropTypes.func
};

export default DriftFilterDropdown;
