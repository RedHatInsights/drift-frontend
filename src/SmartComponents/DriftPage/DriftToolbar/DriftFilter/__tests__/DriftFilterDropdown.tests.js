import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DriftFilterDropdown from '../DriftFilterDropdown';

describe('DriftFilterValue', () => {
    let props;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [ init, setState ]);

    beforeEach(() => {
        props = {
            filterType: 'Fact name',
            toggleFilterType: jest.fn()
        };
    });

    it('should render with fact name', () => {
        const wrapper = shallow(
            <DriftFilterDropdown { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with state filter', () => {
        props.filterType = 'State';

        const wrapper = shallow(
            <DriftFilterDropdown { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
