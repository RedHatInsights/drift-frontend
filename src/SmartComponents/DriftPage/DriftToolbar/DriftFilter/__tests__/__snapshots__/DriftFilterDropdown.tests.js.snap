// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriftFilterValue should render with fact name 1`] = `
<Dropdown
  className="comparison-filter-dropdown-width"
  dropdownItems={
    Array [
      <DropdownItem
        data-ouia-component-id="fact-name-filter"
        onClick={[Function]}
      >
        Fact name
      </DropdownItem>,
      <DropdownItem
        data-ouia-component-id="state-filter"
        onClick={[Function]}
      >
        State
      </DropdownItem>,
    ]
  }
  isOpen={false}
  ouiaId="drift-filter-dropdown"
  toggle={
    <DropdownToggle
      icon={
        <FilterIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      onToggle={[Function]}
      ouiaId="drift-filter-toggle"
    >
      Fact name
    </DropdownToggle>
  }
/>
`;

exports[`DriftFilterValue should render with state filter 1`] = `
<Dropdown
  className="comparison-filter-dropdown-width"
  dropdownItems={
    Array [
      <DropdownItem
        data-ouia-component-id="fact-name-filter"
        onClick={[Function]}
      >
        Fact name
      </DropdownItem>,
      <DropdownItem
        data-ouia-component-id="state-filter"
        onClick={[Function]}
      >
        State
      </DropdownItem>,
    ]
  }
  isOpen={false}
  ouiaId="drift-filter-dropdown"
  toggle={
    <DropdownToggle
      icon={
        <FilterIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      onToggle={[Function]}
      ouiaId="drift-filter-toggle"
    >
      State
    </DropdownToggle>
  }
/>
`;
